src/hotspot/cpu/x86/c1_LIRAssembler_x86.cpp

```diff
@@ -3525,18 +3525,27 @@ void LIR_Assembler::emit_load_klass(LIR_OpLoadKlass* op) {
   Register obj = op->obj()->as_pointer_register();
   Register result = op->result_opr()->as_pointer_register();
 
-  CodeEmitInfo* info = op->info();
-  if (info != NULL) {
-    add_debug_info_for_null_check_here(info);
-  }
```

Comment:
Wait, so this thing would not be executed for `!_LP64` anymore?

Reply:
Indeed! Good catch! Will fix before push.